avocado/utils/config_change_validation.py

```diff
@@ -0,0 +1,205 @@
+"""
+
+SUMMARY
+------------------------------------------------------
+Config change validation: Utility for testing config file changes.
+
+Copyright: 2012 Intra2net AG
+
+
+INTERFACE
+------------------------------------------------------
+
+"""
+
+import os
+import shutil
+
+# avocado utilities
+from avocado.utils import process
+
+
+def get_temp_file_path(file_path):
+    """
+    Generates a temporary filename.
+
+    :param str file_path: file path prefix
+    :returns: appended file path
+    :rtype: str
+    """
+    return file_path + '.tmp'
```

Comment:
Is there a reason for enforcing the temporary file name location (and name)?  The best practice I try to follow when a temporary file name only is needed, is to create a truly temporary and unpredictable temporary directory, and then create your file in it.

Reply:
Agreed on the best practice, the code is at least some years old and this part hasn't been updated ever since on our side. However, the problem is that we are reusing the temporary file names throughout the module so we can't generate new names upon each invocation. One option would be to store the file name but on module level this would imply a use of a global variable. Do you have a better suggestion of what we could do perhaps?